lib/from_openstudio/material/opaque_no_mass.rb

```diff
@@ -42,18 +42,51 @@ def self.from_material(material)
         # set hash values from OpenStudio Object
         hash[:identifier] = material.nameString
         hash[:r_value] = material.thermalResistance
-        hash[:roughness] = material.roughness
-        # check if boost optional object is empty
-        unless material.thermalAbsorptance.empty?
-          hash[:thermal_absorptance] = material.thermalAbsorptance.get
+        # Roughness is a required property for OS MasslessOpaqueMaterial but isn't a listed property
+        # for OS AirGap
+        begin
+          hash[:roughness] = material.roughness
+        rescue
         end
-        # check if boost optional object is empty
-        unless material.solarAbsorptance.empty?
-          hash[:solar_absorptance] = material.solarAbsorptance.get
+        if hash[:roughness].nil?
+          # If property doesn't exist, use default value from schema
+          hash[:roughness] = defaults[:roughness][:default]
         end
         # check if boost optional object is empty
-        unless material.visibleAbsorptance.empty?
-          hash[:visible_absorptance] = material.visibleAbsorptance.get
+        begin
+          # check if boost optional object is empty
+          if !material.thermal_absorptance.empty? hash[:thermal_absorptance] = material.thermalAbsorptance.get
```

Comment:
For some reason, for the AirGap OS material the material.thermalAbsorptance, material.solarAbsorptance and material.visibleAbsorptance returns 0.0, even though the property isnt defined for it in OpenStudio. In this case the unless material.property.empty? statement was throwing an error and needed begin rescue statements.

Reply:
Interesting. That seems like a bug in OpenStudio SDK but I agree the right approach for now is just to work around it.  I think the current begin/rescue might be too implicit and generic that it could fail (or fail silently) after a fix is merged into OpenStudio SDK.

Couldn't we do just one check to see if the material isn't an AirGap and, if so, then we try to get the roughness, absorptances, etc.?